openedx/features/enterprise_support/enrollments/tests/test_utils.py

```diff
@@ -0,0 +1,199 @@
+        with self.assertRaises(UserDoesNotExistException):
+            lms_enroll_user_in_course(
+                None,
+                CourseKey.from_string(COURSE_STRING),
+                'verified'
+            )
+
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.api.add_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.api.get_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.add_user_to_course_cohort')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.User')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.transaction')
+    def test_course_enrollment_error_raises(self,
+                                            mock_tx,
+                                            mock_user_model,
+                                            mock_add_user_to_course_cohort,
+                                            mock_get_enrollment_api,
+                                            mock_add_enrollment_api,
+                                            ):
+        enrollment_response = {'mode': 'verified', 'is_active': True}
+        username = 'test'
+        course_id = CourseKey.from_string(COURSE_STRING)
+        mode = "verified"
+
+        mock_add_enrollment_api.side_effect = CourseEnrollmentError("test")
+        mock_tx.return_value.atomic.side_effect = None
+
+        mock_get_enrollment_api.return_value = enrollment_response
+
+        a_user = {'id': 1223, 'username': username}
+        mock_user_model.return_value.objects.return_value.get.return_value = a_user
+
+        with self.assertRaises(CourseEnrollmentError):
+            lms_enroll_user_in_course(username, course_id, mode)
+            mock_add_user_to_course_cohort.assert_not_called()
+            mock_get_enrollment_api.assert_called_once()
+
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.api.add_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.api.get_enrollment')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.add_user_to_course_cohort')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.User')
+    @mock.patch('openedx.features.enterprise_support.enrollments.utils.transaction')
+    def test_course_group_error_raises(self,
+                                       mock_tx,
+                                       mock_user_model,
+                                       mock_add_user_to_course_cohort,
+                                       mock_get_enrollment_api,
+                                       mock_add_enrollment_api,
+                                       ):
+        enrollment_response = {'mode': 'verified', 'is_active': True}
+        username = 'test'
+        course_id = CourseKey.from_string(COURSE_STRING)
+        mode = "verified"
```

Comment:
nit: Is there a reason not to just define these (enrollment_response, username, course_id, and mode) at the class level? You keep having to define them in every test and they are always exactly the same, so why not just do it once?

Reply:
yeah good point, why do it over and over! I will refactor this

Reply:
```
COURSE_ID = CourseKey.from_string(COURSE_STRING)
USERNAME = 'test'
USER_ID = 1223
```
You've defined them at the top now, which is great, but you still aren't using them here or anywhere else. So this isn't really addressed.

Reply:
The above still applies. You made the variables, which is great, but didn't use them (kind of surprised the style error of unused variables didn't get thrown, maybe it was implemented in one spot but not in a few of the others)